Gems/Atom/RHI/DX12/Code/Source/RHI/Scope.cpp

```diff
@@ -101,19 +101,13 @@ namespace AZ
                  D3D12_RESOURCE_STATE_COPY_DEST |
                  D3D12_RESOURCE_STATE_COPY_SOURCE);
 
-            constexpr D3D12_RESOURCE_STATES VALID_GRAPHICS_QUEUE_RESOURCE_STATES =
-                (D3D12_RESOURCE_STATES)DX12_RESOURCE_STATE_VALID_API_MASK;
-
             switch (GetHardwareQueueClass())
             {
             case RHI::HardwareQueueClass::Graphics:
-                return RHI::CheckBitsAll(VALID_GRAPHICS_QUEUE_RESOURCE_STATES, state);
+                return true;
 
             case RHI::HardwareQueueClass::Compute:
                 return RHI::CheckBitsAll(VALID_COMPUTE_QUEUE_RESOURCE_STATES, state);
```

Comment:
Hmm. We still set D3D12_COMMAND_LIST_TYPE_COPY and do create a separate hw queue for copy although no workload is going to it atm (but can in future as it is set by the pipeline in RPI). Removing this case means that we do not support RHI::HardwareQueueClass::Copy at all for DX12 which seems weird. We should in theory create a VALID_COPY_QUEUE_RESOURCE_STATES (similar to compute) and use that as a check here.

Reply:
I am not sure if I missed it, but as far as I could tell, the `DX12_RESOURCE_STATE_COPY_QUEUE_BIT` bit is currently only activated in `FrameGraphCompiler::GetResourceState()`, which only occurs in `FrameGraphCompiler::CompileBuffer/ImageBarriers()` and the `Scope::IsStateSupportedByQueue()` is the only portion of the code that checks for this bit.
In `CompileBufferBarriers()` this combination does not occur, only in `CompileImageBarriers()` and there, in the one place it could happen, the check is now extra to the call
```cpp
if (scopeAfter.IsStateSupportedByQueue(transition.StateBefore) && (scopeAfter.GetHardwareQueueClass() == RHI::HardwareQueueClass::Copy ? copyQueueState : scopeAfter.IsStateSupportedByQueue(transition.StateAfter)))
```

Am I missing something here, should that not result in the same outcome as before?

Reply:
Are any changes requested? otherwise this PR is ready to merge

Reply:
I apologize for the delay in response. The issue is that dx12 backend creates a copy queue within CommandQueueContext::Init which then calls CreateCommandQueue with D3D12_COMMAND_LIST_TYPE_COPY to create the hw queue. So RHI is pumping three queues (which includes copy queue) every frame even though the main pipeline is **not** queuing any workload on the copy queue at the moment so any functionality related to it is untested. 

This function (in theory) should handle queries for all three queues. After this change we are basically saying that IsStateSupportedByQueue will always return false for copy queue because the copy queue functionality is not implemented. We haven't deprecated copy hw queue but have now removed implementation related to it. If the implementation is wrong we should either implement the correct implementation (possible recommendation provided in my previous comment) or we leave a 'TODO' with a warning/assert here in case this function is ever invoked with copy queue.

Reply:
I see what you are getting at, I will take a closer look and then amend this PR appropriately.